File: models/VmWord.cs
﻿using System;
using System.Collections.Generic;

namespace EnglishTraining
{
    public class VmWord
    {
        public int Id { get; set; }
        public string Name_en { get; set; }
        public string Name_ru { get; set; }
        public bool FourDaysLearnPhase { get; set; }
        public int LearnDay { get; set; }
        public int RepeatIterationNum { get; set; }
        public DateTime NextRepeatDate { get; set; }
        public int DailyReapeatCountForEng { get; set; }
        public int DailyReapeatCountForRus { get; set; }
    }

    public class VmWordWithDictors : VmWord
    {
        public IList<VmDictor> Dictors_en { get; set; }
        public IList<VmDictor> Dictors_ru { get; set; }
        public IList<VmCollocation> Collocation { get; set; }
    }
}

File: Frontend/src/app/models/VmWord.ts
import { VmCollocation } from './VmCollocation';
import { VmDictor } from './VmDictor';
import { Dictionary } from '../interfaces/Dictionary';

export interface VmWord {
    Id: number;
    learnDay: Dictionary<number>;
    fourDaysLearnPhase: Dictionary<boolean>;
    repeatIterationNum: Dictionary<number>;
    nextRepeatDate: Dictionary<Date>;
    dailyReapeatCount: Dictionary<number>;
    langDictionary: Dictionary<string>;
    dictors: Dictionary<VmDictor[]>;
    // TODO: find out if it should be Dictionary
    // collocation: Dictionary<VmCollocation[]>;
    collocation: VmCollocation[];
}
